contrib/dimfeld/httptreemux.v5/httptreemux.go

```diff
@@ -112,18 +112,26 @@ func getRoute(router *httptreemux.TreeMux, w http.ResponseWriter, req *http.Requ
 	if !found {
 		return "", false
 	}
+	routeLen := len(route)
+	trailingSlash := route[routeLen-1] == '/' && routeLen > 1
 
-	// Check for redirecting route due to trailing slash for parameters.
-	// The redirecting behaviour originates from httptreemux router.
-	if lr.StatusCode == http.StatusMovedPermanently && strings.HasSuffix(route, "/") {
+	// Retry the population of lookup result parameters.
+	// If the initial attempt to populate the parameters fails, clone the request and modify the URI and URL Path.
+	// Depending on whether the route has a trailing slash or not, it will either add or remove the trailing slash and retry the lookup.
+	if routerRedirectEnabled(router) && isSupportedRedirectStatus(lr.StatusCode) && lr.Params == nil {
 		rReq := req.Clone(req.Context())
-		rReq.RequestURI = strings.TrimSuffix(rReq.RequestURI, "/")
-		rReq.URL.Path = strings.TrimSuffix(rReq.URL.Path, "/")
-
-		lr, found = router.Lookup(w, rReq)
-		if !found {
-			return "", false
+		if trailingSlash {
+			// if the route has a trailing slash, remove it
+			rReq.RequestURI = strings.TrimSuffix(rReq.RequestURI, "/")
+			rReq.URL.Path = strings.TrimSuffix(rReq.URL.Path, "/")
+		} else {
+			// if the route does not have a trailing slash, add one
+			rReq.RequestURI = rReq.RequestURI + "/"
+			rReq.URL.Path = rReq.URL.Path + "/"
```

Comment:
Does this mean we also redirect when there is no trailing slash?

Reply:
Yes. That's correct. This is the case we missed originally.